### DISCUSSION TITLE
Add a SkeletonMask Resource for AnimationTree filtering

### DISCUSSION THREAD

**Discussion by Jesusemora** [godotengine/godot-proposals#7896] [Editor]
some nodes in the AnimationTree have a filter option that allows you to select which bone transforms will be overridden/added to another animation.

but if you use this option, you have to manually select each bone in the skeleton every time.

I propose to replace or extend it with a new SkeletonMask resource. this way you can set up your mask once and then assign it to the filter, allowing you to reuse it multiple times.

This is the skeleton filter right now:
![filter](https://docs.godotengine.org/en/stable/_images/animtree6.png)

I'm thinking of something like this:
![SkeletonMask](https://github.com/godotengine/godot-proposals/assets/32273722/50c6661e-c634-4dea-8b25-8cdcabf6729a)

and move the bone selection inside the mask.

**Comment by Glitshy** [comment]
Just came across this in the context of SekeletonModifier3D, for which this would be great as well,

**Comment by MyMilkBiscuit** [comment]
This would be very useful for setting filters via code. The current method requires that you set every bone individually with a path reference which isn't ideal, and doesn't allow you to set up bone filters visually the way you can in the AnimationTree.
